### DISCUSSION TITLE
Logo overlay (png image) with OME Transcoder

### DISCUSSION THREAD

**Discussion by TheRede** [OvenMediaLabs/OvenMediaEngine#1276] [Help]
Thanks for the nice product you have built over the years, been here 2 years ago and the progress is huge.
At the moment one of the moments stopping me switching from Flussonic is logo overlays, there are times when we need to burn in/overlay a logo of the service provider or some partner to the video, do you have it in plans or maybe it is already in motion?

**Comment by getroot** [comment]
This is a possible task, but I don't know how many people would want this feature. I think most users will want to do this in their encoder.

If anyone wants this feature, please comment with a use case.

**Reply by zijianhewwork** [reply]
I’d also really like to see this feature added. In my use case, we need to burn in a watermark/logo directly into the stream at the server side (before delivery), not just overlay in the web player via HTML/CSS. This is important because:

- We often redistribute streams to third-party platforms where we don’t control the player.
- Some partners require their branding/watermark to be embedded at the stream source level for compliance.
- Doing this on the encoder side isn’t always practical when multiple partner logos are needed depending on the output.

Having native logo overlay support in the server would make the workflow much more flexible and reduce load on encoders.

Is there currently any recommended workaround for this (e.g., integrating with FFmpeg filters or another approach)?

**Reply by bchah** [reply]
+1000 for this!

**Reply by bchah** [reply]
To be fair, this is already a feature in OvenMediaEngine Enterprise so it is there for the purchasing if you really need it. But I also agree it would be a good feature for everybody 😀

**Comment by getroot** [comment]
Unfortunately, the Overlay feature was rolled out first in the Enterprise edition, as it is closely tied to branding and positioned as a commercial feature.

https://docs.enterprise.ovenmediaengine.com/guide/enhancement/image-overlay

While we cannot disclose further details due to NDAs with our customers, it had to be the Enterprise users who were given early access. I believe this feature may also become available in the Open-Source in the future.

For those in our open-source community who wish to try it now, we will gladly provide an Enterprise demo. Please contact us at contact@airensoft.com
